### ISSUE TITLE
empty result folder(run_exp)

### ISSUE THREAD

**Issue by juliaa123** [hongzimao/pensieve#114]
hi,
I run run_all_traces.py in folder run_exp, the result folder includes empty files, i follow the suggested solutions at issue #98,  but still the same problem. I think the problem in URL address which used in run-video code , can you explain how select this link and why use it, and how change it ??
note , the chrom_rerty_log is not includes any error.

thanks for your time.

**Comment by hongzimao** [comment]
If you copy over the html files to the apache server folder (done in the setup step here: https://github.com/hongzimao/pensieve/blob/master/setup.py#L44-L47), you should be able to watch the video with your browser. See if you can load the video by typing in the localhost and url to the video in a regular chrome?

The video might only play the first chunk --- this is because the second and future chunks need an ABR decision from the ABR server. In this case, if you run a python ABR server, similar to https://github.com/hongzimao/pensieve/blob/master/run_exp/run_video.py#L54, you should be able to see the ABR server making every ABR decision (you can print it) and the video client receive the bitrate for each chunk (you can log it in javascript) and the video keeps playing. 

Hope these help!

**Comment by juliaa123** [comment]
Thanks for reply me,
I did what you suggested, and the video is playing but without autoplay, i started the video manually after running the apache server.

**Comment by hongzimao** [comment]
This might help: #73

**Comment by juliaa123** [comment]
still empty, i check all the software environment, google chroom V56, driver 2.27
and i change the URL to local host , when i change it and run the RL server manually based #98 issue  , i see log file with data in result folder in the RL server folder , but when i run all-traces direct , the result folder is empty in RL server folder and in Run_exp folder.

now  , i do not have any idea how to solve this problem.

please help me,
thanks for your time

**Comment by hongzimao** [comment]
So you can watch the video manually (video loads, new chunks are downloaded, ABR server runs nominally, etc.) but when you run everything together with run_all_traces, it doesn't log anything, is that the issue? 

So run_all_traces starts a virtual display (with chromium). If you really want to debug, you can ask the virtual display to screenshot to see what's the video state (does it load the first chunk, is it the auto play issue, etc.)

Did #73 not help with the auto play issue?

Also, one thing you can try is this repo https://github.com/park-project/park. Try using the `abr` environment. This codebase has slightly more robust support for the ABR environment.

**Comment by ali1hammoud** [comment]
![aa](https://user-images.githubusercontent.com/37132652/91645931-979ba800-ea52-11ea-9c75-eb36f95f1abd.png)
here is the screenshot.
and this is the error : 
Message: u'timeout: cannot determine loading status\nfrom timeout: Timed out receiving message from renderer: -0.329\n  (Session info: chrome=56.0.2924.87)\n  (Driver info: chromedriver=2.27.440175 (9bc1d90b8bfa4dd181fbbf769a5eb5e575574320),platform=Linux 4.15.0-112-generic x86_64)' 

I think the page doesn't load?
thank you.

**Comment by ali1hammoud** [comment]
![aa1](https://user-images.githubusercontent.com/37132652/91646113-6cb25380-ea54-11ea-8fee-ab50f739441c.png)

I did everything in issue 98, And I saw that the video doesn't load even if I bypass mahimahi
So the video didn't start automatically, I pressed the play button, And the video didn't load at all.

#I WROTE here because I have the same problem.

**Comment by hongzimao** [comment]
@ali1hammoud This is strange. If you bypass mahimahi the video should play at least for the first chunk (it doesn't ask the ABR server for the bitrate). I see you use firefox as the browser. Can you try chrome too? Make sure the chromedriver and selenium version is correct. Can you also try the abr environment in repo https://github.com/park-project/park. This codebase has slightly more robust support for the ABR environment.

**Comment by chutoutian** [comment]
I have the same problem. I run the rl_server_no_training.py in /rl_server , then I run the run_all_traces.py in /run_exp. The video can play completely. But /run_exp/results/ is empty .

**Comment by ry4nzhu** [comment]
I experience similar problem here, I changed the video to my own video chunks, the player only request the first 6 chunks and never send any request to the ABR server at localhost:8333 again even the buffer is drained.

**Comment by hongzimao** [comment]
@ry4nzhu Did you manage to get the original video to work? If that works, can you try just swapping out the original video with your own video chunks (rename them with the same name), that should load the video.

**Comment by ry4nzhu** [comment]
Yes. I got the original video to work. Does it matter if I use my own mpd file and put all the video chunk in one folder in the apache server? I can try to swapping out the original video with my own video chunks, but I doubt if this is different from changing the mpd file in myindex_${algorithm}.html and put all the video file in a separate folder..

**Comment by hongzimao** [comment]
Sorry for replying late. I think you should swap out the mpd file too. You can put the videos in the apache server, then use the browser to go to the URL https://github.com/hongzimao/pensieve/blob/master/run_exp/run_video.py#L39. You can check if your videos are loading and playing. Hope this helps.

**Comment by SunTingxin** [comment]
hi
I have  run  https://github.com/hongzimao/pensieve/blob/master/run_exp/run_video.py#L54 ,but it still just plays the first chunk. How can I know whether the ABR  server works？
![2021-05-26 10-06-44 的屏幕截图](https://user-images.githubusercontent.com/40429606/119591970-297de900-be0a-11eb-815f-0aa385628cd9.png)
![2021-05-26 10-06-49 的屏幕截图](https://user-images.githubusercontent.com/40429606/119591994-34d11480-be0a-11eb-88c0-f8b4ba37316d.png)
